components/filter/filter.js

```diff
@@ -301,8 +302,20 @@ class Filter extends LocalizeCoreElement(RtlMixin(LitElement)) {
 		`;
 	}
 
-	_dispatchChangeEvent(eventDetail) {
-		this.dispatchEvent(new CustomEvent('d2l-filter-change', { bubbles: true, composed: false, detail: eventDetail }));
+	_dispatchChangeEvent(eventKey, eventDetail) {
+		this._changeEventsToDispatch.set(eventKey, eventDetail);
+
+		if (!this._changeEventTimeout) {
+			this._changeEventTimeout = setTimeout(() => {
+				this.dispatchEvent(new CustomEvent('d2l-filter-change', {
+					bubbles: true,
+					composed: false,
+					detail: { changes: Array.from(this._changeEventsToDispatch.values()) }
+				}));
+				this._changeEventsToDispatch = new Map();
+				this._changeEventTimeout = null;
+			}, 200);
```

Comment:
This 200 value was based on what "felt" good - nothing real scientific.  This was about the fastest I could click two separate filters to have them grouped, or clicking/unclicking the same option about two or three times.  Longer would be more performant (especially if someone keeps toggling the same option), but might have a noticeable delay on the user side before anything starts loading.

The _main_ purpose here is to batch the clear/select all case anyways, so we could go lower if we were worried.